### ISSUE TITLE
[FEATURE] Support multiple datasources in the "Roles" tab

### ISSUE THREAD

**Issue by derek-ho** [opensearch-project/security-dashboards-plugin#1798]
**Is your feature request related to a problem?**
Support multiple datasources functionality in the "Roles" tab. This will be a more meaty sub task since this tab contains links to other pages.

**Acceptance criteria**
- Data source picker shared component is present on the page with a sensible UX
- Data source selections that should be persisted across tabs is accurately persisted
- Selected datasource prior to landing on this page is correctly populated (if applicable), and defaults are selected if not applicable
- Create role button leads to a page with either the cluster picker in read only mode, or writeable mode, so it is clear which datasource the role is being created for. 
- View role button leads to the same as above
- Edit role button leads to the same as above
- Duplicate role button leads to the same as above
- Delete role modal makes it clear which datasource the role deletion is happening for (is this p0?)

**What solution would you like?**
Support multiple datasources in the "Roles" tab. 

**What alternatives have you considered?**
None

**Do you have any additional context?**
Depends on #1795 to have data source picker available.

**Comment by zhongnansu** [comment]
@derek-ho @cwperks  Is this completed? can we close?

**Comment by derek-ho** [comment]
Code is in feature branch closing

**Comment by DarshitChanpura** [comment]
@zhongnansu This was merged into the feature branch. I'm closing this. However, this has yet to be merged into main.
